inc/feedwriter.php

```diff
@@ -6,8 +6,8 @@ class FeedWriter {
 	public function __construct($id) {
 		$this->feedID = $id;
 		$this->baseURL =
-			'http://' .
-			$_SERVER['HTTP_HOST'] .
+			http_https() .
+			$_SERVER['SERVER_NAME'] .
```

Comment:
Ditto, but I read the [server array doc](http://www.php.net/manual/en/reserved.variables.server.php) in this way:
- `HTTP_HOST` is reading from a header field (thus set by the client itself) witch might be undefined in certain circumstances (I had it when using some old cURL libs)
- `SERVER_NAME` is set by the http server itself, and therefore always populated.

> You should now however realize that the one is a client-controlled value which may thus not be reliable for use in business logic and the other is a server-controlled value which is more reliable. You however need to ensure that the webserver in question has the SERVER_NAME correctly configured.
https://stackoverflow.com/a/2297421